src/BitStringAddresses/bosefs.jl

```diff
@@ -331,10 +331,57 @@ end
 function hopnextneighbour(b::SingleComponentFockAddress, i)
     src = find_occupied_mode(b, (i + 1) >>> 0x1)
     dst = find_mode(b, mod1(src.mode + ifelse(isodd(i), 1, -1), num_modes(b)))
-
     new_b, val = excitation(b, (dst,), (src,))
     return new_b, val
 end
+"""
+    new_address, product = hopnextneighbour(add, chosen, Val(boundary_condition))
+
+Compute the new address of a hopping event for the Hubbard model. Returns the new
+address and the square root of product of occupation numbers of the involved modes
+multiplied by a term consistent with boundary condition.
+
+The off-diagonals are indexed as follows:
+
+* `(chosen + 1) ÷ 2` selects the hopping site.
+* Even `chosen` indicates a hop to the left.
+* Odd `chosen` indicates a hop to the right.
+* Boundary conditions are periodic, pi-twisted and hard-wall.
```

Comment:
This makes it seem like all three are true at the same time. How about
```suggestion
* Boundary conditions can be periodic, pi-twisted and hard-wall.
```